base/set.jl

```diff
@@ -17,6 +17,46 @@ Set() = Set{Any}()
 Construct a [`Set`](@ref) of the values generated by the given iterable object, or an
 empty set. Should be used instead of [`BitSet`](@ref) for sparse integer sets, or
 for sets of arbitrary objects.
+
+# Examples
+```jldoctest
+julia> animals = Set{String}(["cat","fish","dog","bird"])
+Set{String} with 4 elements:
+  "fish"
+  "cat"
+  "bird"
+  "dog"
+```
+
+If the element type is not given, it is computed by intersecting the types of the arguments:
+```jldoctest
+julia> animals = Set(["cat","fish","dog","bird"])
+Set{String} with 4 elements:
+  "fish"
+  "cat"
+  "bird"
+  "dog"
+
+julia> typeof(animals)
+Set{String}
+```
+
+A common way of incrementally creating a `Set` is to first create an empty set with `Set()` and then add the elements using the `push!` function:
+```jldoctest
+julia> int_container = Set{Int64}() # type specified
+Set{Int64} with 0 elements
+
+julia> push!(int_container, 2)
+Set{Int64} with 1 element:
+  2
+
+julia> any_container = Set() # type not specified
+Set{Any} with 0 elements
+
+julia> push!(any_container, "hi!")
```

Comment:
You could perhaps `push!` a number into this one as well to show that you can have elements of different types in a `Set{Any}`.